requests/models.py

```diff
@@ -907,7 +905,16 @@ def json(self, **kwargs):
                     # and the server didn't bother to tell us what codec *was*
                     # used.
                     pass
-        return complexjson.loads(self.text, **kwargs)
+
+        try:
+            return complexjson.loads(self.text, **kwargs)
+        except JSONDecodeError as e:
+            # Catch JSON-related errors and raise as requests.JSONDecodeError
+            # This aliases json.JSONDecodeError and simplejson.JSONDecodeError
+            if is_py2: # e is a ValueError
+                raise RequestsJSONDecodeError()
+            else:
+                raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
 
```

Comment:
Giving the user 0 useful information is unacceptable. Why can't we use `e.message` there?

```py
Python 2.7.18 (default, May 19 2021, 00:00:00)
[GCC 11.1.1 20210428 (Red Hat 11.1.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> try:
...   json.loads("foo")
... except ValueError as ve:
...   pass
...
>>> dir(ve)
['__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__getitem__', '__getslice__', '__hash__', '__init__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__unicode__', 'args', 'message']
>>> ve.message
'No JSON object could be decoded'
>>> ve.args
('No JSON object could be decoded',)
>>>
```

Reply:
I added that. Workflows passed. Thanks for mentioning this!